test-data/unit/check-namedtuple.test

```diff
@@ -185,3 +185,35 @@ class C(B): pass
 B(1).b
 C(2).b
 [builtins fixtures/property.py]
+
+[case testNamedTupleAsDict]
+from collections import namedtuple
+
+X = namedtuple('X', ['x', 'y'])
+x = None  # type: X
+a = x._asdict()
```

Comment:
Maybe we need to use a totally different approach where mypy merges the
synthesized class with the template imported from collections.abc or typing
(both of which currently contain dummy definitions that aren't even
classes).